data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexConfigurationTests.java

```diff
@@ -84,12 +88,16 @@ public void testValidCustom_from_s3() {
         final String testTemplateFilePath = "s3://folder/file.json";
         final String testS3AwsRegion = "us-east-2";
         final String testS3StsRoleArn = "arn:aws:iam::123456789:user/user-role";
+        final String fileContent = "{}";
 
         final S3Client s3Client = mock(S3Client.class);
-        when(s3Client.getObject(any(GetObjectRequest.class))).thenReturn(null);
 
-        final GetObjectAttributesResponse getObjectAttributesResponse = GetObjectAttributesResponse.builder().objectSize(100L).build();
-        when(s3Client.getObjectAttributes(any(GetObjectAttributesRequest.class))).thenReturn(getObjectAttributesResponse);
+        final InputStream fileObjectStream = IOUtils.toInputStream(fileContent, StandardCharsets.UTF_8);
+        final ResponseInputStream<GetObjectResponse> fileInputStream = new ResponseInputStream<>(
+                GetObjectResponse.builder().contentLength(1_000_000L).build(),
```

Comment:
`1_000_000` is a magic number. You can rename this to `SIZE_SMALLER_THAN_THRESHOLD`.